File: app/src/main/java/com/example/sergiobelda/photoeditor/editableimageview/figures/Square.java
package com.example.sergiobelda.photoeditor.editableimageview.figures;

public class Square implements Polygon {
    private float x, y;
    private double side;
    private float color;
    private float rotation;

    public Square(float x, float y, double side, int color) {
        this.x = x;
        this.y = y;
        this.side = side;
        this.color = color;
        this.rotation = 0;
    }

    @Override
    public float getX() {
        return x;
    }

    @Override
    public void setX(float x) {
        this.x = x;
    }

    @Override
    public float getY() {
        return y;
    }

    @Override
    public void setY(float y) {
        this.y = y;
    }

    public double getSide() {
        return side;
    }

    public void setSide(double side) {
        this.side = side;
    }

    public float getColor() {
        return color;
    }

    public void setColor(float color) {
        this.color = color;
    }

    public float getSize(){
        return (float) getSide();
    }

    @Override
    public void setSize(float size) {
        this.setSide(size);
    }

    public float getRotation() {
        return rotation;
    }

    public void setRotation(float rotation) {
        this.rotation = rotation;
    }
}

File: app/src/main/java/com/example/sergiobelda/photoeditor/editableimageview/figures/Square.kt
package com.example.sergiobelda.photoeditor.editableimageview.figures

open class Square(private var x: Float, private var y: Float, var side: Float, color: Int) : Polygon {
    private var color: Float = 0.toFloat()
    var rotation: Float = 0.toFloat()

    init {
        this.color = color.toFloat()
        this.rotation = 0f
    }

    override fun getX(): Float {
        return x
    }

    override fun setX(x: Float) {
        this.x = x
    }

    override fun getY(): Float {
        return y
    }

    override fun setY(y: Float) {
        this.y = y
    }

    override fun getColor(): Float {
        return color
    }

    fun setColor(color: Float) {
        this.color = color
    }

    override fun getSize(): Float {
        return side.toFloat()
    }

    override fun setSize(size: Float) {
        this.side = size
    }
}